crates/bin/node/src/grpc_service.rs

```diff
@@ -306,7 +334,35 @@ impl Node for GrpcState {
         let amount = Amount::from(mint_quote_request.amount);
         let unit = Unit::from_str(&mint_quote_request.unit).map_err(ParseGrpcError::Unit)?;
 
+        let request_hash = hash_mint_quote_request(mint_quote_request.clone());
+        let cache_key = format!("{:?}{}", method, request_hash);
+        let use_cache = self
+            .should_cache_endpoint(HttpMethod::Post, Path::Mint(method.clone()))
+            .await;
+        if use_cache {
+            // Try to get from cache first
+            if let Some(cached_response) = self.response_cache.clone().get(&cache_key) {
+                if let CachedResponse::MintQuote(mint_quote_response) = cached_response {
+                    return Ok(Response::new(mint_quote_response));
+                }
+            }
+        }
```

Comment:
Can you factorize this code? make it a function and match it's output to decide whether to return early or continue the execution